memgpt/presets.py

```diff
@@ -1,41 +1,89 @@
 from .prompts import gpt_functions
 from .prompts import gpt_system
 
-DEFAULT_PRESET = "memgpt_chat"
-preset_options = [DEFAULT_PRESET]
+from typing import List, Optional, Tuple
+from dataclasses import dataclass, field
 
+DEFAULT_PRESET = "memgpt_chat"
+preset_options = [DEFAULT_PRESET]  # TODO: eventually remove
 SYNC_CHAT = "memgpt_chat_sync"  # TODO: remove me after we move the CLI to AgentSync
 
 
-def use_preset(preset_name, agent_config, model, persona, human, interface, persistence_manager):
+@dataclass
+class Preset:
+    name: str = None
+    pretty_name: str = None
+    functions: Tuple[str] = ()
+    sync: bool = False  # TODO: remove as a preset
+
+
+@dataclass
+class DefaultPreset(Preset):

```

Comment:
I think this refactor is generally the right idea, however I wonder what the best way is to support users adding their own presets, eg: https://github.com/cpacker/MemGPT/pull/282

I feel like adding presets should ideally be as easy as adding personas/humans or should have a similar workflow, where you create a text file and put it in a folder.

Thoughts?